backend/cmm_helpers.ml

```diff
@@ -2334,6 +2334,15 @@ let ptr_offset ptr offset dbg =
 let direct_apply lbl ty args (pos, _mode) dbg =
   Cop (Capply (ty, pos), Cconst_symbol (lbl, dbg) :: args, dbg)
 
+let split_arity_for_apply arity args =
+  let max_arity = Lambda.max_arity () in
+  if List.compare_length_with arity max_arity > 0
+  then
+    let a1, a2 = Misc.Stdlib.List.split_at max_arity arity in
+    let args1, args2 = Misc.Stdlib.List.split_at max_arity args in
```

Comment:
This is incorrect if there are unboxed tuples, as `args` will not have the same length as `arity` in this case.

Reply:
I don't know much about unboxed tuples. In that case what would be the correct way to split args?